snapcraft/internal/build_providers/errors.py

```diff
@@ -126,13 +127,28 @@ def __init__(
         )
 
 
+class ProviderInvalidBaseError(_SnapcraftException):
+    def __init__(self, *, provider_name: str, build_base: str) -> None:
+        self.provider_name = provider_name
+        self.build_base = build_base
+
+    def get_brief(self) -> str:
+        return f"The {self.provider_name!r} does not have an environment setup to build for base {self.build_base!r}"
```

Comment:
```suggestion
f"The {self.provider_name!r} build provider does not currently support base {self.build_base!r}"
```

Reply:
done, but ignored "build" in provider for consistency with the other messages and did not add "currently" to not give a sense of "it is coming"